### PR TITLE
3893890:Issue:Exclusion list CI testing for PDR plugin

### PR DESCRIPTION
**Pull request by vg12345**

### CODE REVIEW

File: plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py [hunk 1]
```diff
@@ -885,7 +886,11 @@ def main_flow(self):
                 t_begin = time.time()
                 self.exclude_list.refresh()
                 self.get_isolation_state()
-                self.logger.info("Retrieving telemetry data to determine ports' states")
+                if not self.test_mode:
+                    self.logger.info("Retrieving telemetry data to determine ports' states")
+                else:
+                    self.logger.info(f"Retrieving test mode telemetry data to determine ports' states: iteration {self.test_iteration}")
+                    self.test_iteration += 1
```
Comment (egershonNvidia) [inline plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py:893]: you dont use the test_iteration variable. you can remove it

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py [hunk 2]
```diff
@@ -46,38 +75,69 @@ def do_GET(self):
...
+# All negaitive tests (tested ports should not be isolated)
+# (iteration, row index, counter name): value
+NEGATIVE_DATA_TEST = {
+    # example, also negative test
+    (1, 0, PHY_SYMBOL_ERROR): 0,
+    
+    # testing blacklist
+    (0, 5, EXCLUDE_PORT_LONG_TIME): 0, # add to blacklist forever
+    (1, 5, LINK_DOWN_COUNTER): 1,
+    (2, 5, LINK_DOWN_COUNTER): 2,      # try trigger isolation issue (should be ignored)
```
Comment (egershonNvidia) [inline plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:121]: the comments are not needed, you wrote above that it blacklist test

**Review by egershonNvidia — COMMENTED** [review]
Have you fixed the CI ? how do you test this blacklist?

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py (hunk 2 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:121]: Different blacklist tests consist of several consecutive steps/operations. The steps are different for different blacklist tests. The comments describe what we are expecting from each step. So, I would leave the comments.

**Review by egershonNvidia — APPROVED** [review]

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py (hunk 2 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:121]: I will :)

**Comment by vg12345** [comment]
bot:retest

**Comment by vg12345** [comment]
bot:retest

**Comment by vg12345** [comment]
bot:retest

**Comment by BennyItkin** [comment]
bot:retest

**Comment by vg12345** [comment]
bot:retest

**Comment by vg12345** [comment]
bot:retest

**Comment by vg12345** [comment]
bot:retest

**Comment by BennyItkin** [comment]
bot:retest

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py [hunk 3]
```diff
@@ -130,31 +193,69 @@ def start_server(port:str,changes_intervals:int, run_forever:bool):
...
         output = [header] + data
         csv_data = '\n'.join([','.join(row) for row in output]) + '\n'
-        endpoint['data'] = csv_data        
+        endpoint['data'] = csv_data
         if not run_forever and ENDPOINT_CONFIG["ITERATION_TIME"] > MAX_ITERATIONS:
             # after all the tests are done, we need to stop the simulator and check the logs
             return
         time.sleep(changes_intervals)
 
+def black_ports_simulation(endpoint):
```
Comment (TalLerner) [inline plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:210]: Please rename the method to excluded_ports_simulation

**Review by drorlevy — APPROVED** [review]

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py (hunk 3 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py]: Done. Some comments are changed too.

**Review by TalLerner — APPROVED** [review]

**Pull request merged**